opensquirrel/ir.py

```diff
@@ -535,6 +552,12 @@ def named_gate(gate_generator: Callable[..., ControlledGate]) -> Callable[..., C
 
 
 def named_gate(gate_generator: Callable[..., Gate]) -> Callable[..., Gate]:
+    def is_qubit_like_annotation(annotation: Any) -> bool:
+        return annotation in (QubitLike, Qubit)
+
+    def is_supports_int_annotation(annotation: Any) -> bool:
```

Comment:
Can we?:

1. Put these functions close to `ANNOTATIONS_TO_TYPE_MAP` .
2. Try to use them anywhere else in the code, at least in this `ir.py` file? There is more code like `annotation in (blah, foo)` around.

Thanks!